projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php

```diff
@@ -2239,6 +2422,16 @@ public function __construct( $attributes, $content = null ) {
 			$attributes['id'] = 'widget-' . $attributes['widget'];
 			// translators: the blog name.
 			$default_subject = sprintf( _x( '%1$s Sidebar', '%1$s = blog name', 'jetpack' ), $default_subject );
+		} elseif ( ! empty( $attributes['block_template'] ) && $attributes['block_template'] ) {
+			$default_to      .= get_option( 'admin_email' );
+			$attributes['id'] = 'block-template-' . $attributes['block_template'];
+			// translators: the blog name.
+			$default_subject = sprintf( _x( '%1$s Block Template', '%1$s = blog name', 'jetpack' ), $default_subject );
+		} elseif ( ! empty( $attributes['block_template_part'] ) && $attributes['block_template_part'] ) {
+			$default_to      .= get_option( 'admin_email' );
+			$attributes['id'] = 'block-template-part-' . $attributes['block_template_part'];
+			// translators: the blog name.
+			$default_subject = sprintf( _x( '%1$s Block Template Part', '%1$s = blog name', 'jetpack' ), $default_subject );
```

Comment:
It would be great if we can modify the title a little bit to keep the old context.

I can see that the user loses context from which page was sent the message.

`/wp-admin/edit.php?post_type=feedback`

In my example, it was saying: `Site Title] Hello World!` before the fix.
After the fix it says: `[Site Title] Block Template` or `[Site Title] Block Template Part`

<img width="933" alt="Screenshot 2022-06-15 at 15 49 24" src="https://user-images.githubusercontent.com/779993/173857249-fc40dc30-ccaa-45e3-b772-39c724f5cda5.png">

Reply:
> It would be great if we can modify the saved title to keep the context.

That would be a great improvement 😄 

Unfortunately, I don't know of a good way to identify a _name_ for the context (aside from inventing our own system). Do you know of any ways? Alternatively, maybe there's a more generic title we could use?

Reply:
> What if we remove `$default_subject` from all those conditions and then use this below?

While we'd have a blog post title, it may or may not be accurate:
* On the homepage, `$post` would be the last post in the main query, unless...
* Anywhere there's a `WP_Query` immediately before the contact form, `$post` would be the last post in that query.

Reply:
@danielbachhuber isn't it how it worked before the change? I've roughly tested that case and it was not using the last post, but the homepage's main $post.

Reply:
> isn't it how it worked before the change? I've roughly tested that case and it was not using the last post, but the homepage's main `$post`.

@wojtekn Oh, fair enough. I've restored the prior (somewhat inaccurate) behavior for all contexts with 888a8ad8fd. Here's what it looks like:

<img width="630" alt="image" src="https://user-images.githubusercontent.com/36432/174083343-750f40b3-fe4f-4bbe-8930-d04625cec0e6.png">

Reply:
@danielbachhuber it looks solid and works well for me in all cases now.